### PR TITLE
fix: assetTree will re-render when displayName prop is changed

### LINKED ISSUE #448: AssetTree: modifications to displayName on runtime doesn't affect the rendering
If I use the AssetTree with the displayName-prop:
```js
<AssetTree displayName={(asset) => `asset.id`} />
```

and I later in my code have to change the `displayName` logic to somehting like:
```js
<AssetTree displayName={(asset) => `asset.name`} />
```

then the component will not be updated on the change. You don't utilize the `componentDidUpdate()` https://reactjs.org/docs/react-component.html#componentdidupdate to trigger a re-render.

### PR DESCRIPTION
**Pull request by ArbenPaloja**
Fixes #448

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-449.gearboxjs.preview.cogniteapp.com

File: src/components/AssetTree/AssetTree.tsx [hunk 1]
```diff
@@ -156,26 +138,24 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
...
-      }
+    const cdfAssetChildren = await this.cursorApiRequest(assetId);
+    if (cdfAssetChildren.length > 0) {
+      treeNode.props.dataRef.children = [...cdfAssetChildren]
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .filter(
+          asset =>
+            asset.parentId && asset.parentId === treeNode.props.dataRef.asset.id
+        )
+        .map(asset => AssetTree.convertToNode(asset));
```
Comment (hernil) [inline src/components/AssetTree/AssetTree.tsx]: This looks like a breaking change together with [line 208](https://github.com/cognitedata/gearbox.js/pull/449/files#diff-ebaf1e7f0d00251574a00e040d59e4adR208) as `converToNode` returns an `AssetTreeNode` object incompatible with the `OnSelectAssetTreeParams` interface.

Comment (fredrik955) [reply src/components/AssetTree/AssetTree.tsx]: @polomani Didn't you fix this?

Comment (polomani) [reply src/components/AssetTree/AssetTree.tsx]: it's here https://github.com/cognitedata/gearbox.js/pull/452

**Pull request merged**